src/main/kotlin/net/djvk/fireflyPlaidConnector2/sync/PolledSyncRunner.kt

```diff
@@ -333,4 +340,16 @@ class PolledSyncRunner(
         terminated.set(true)
         mainJob.cancel()
     }
-}
\ No newline at end of file
+
+    companion object {
+        fun emptyPlaidResponse(): TransactionsSyncResponse =
+            TransactionsSyncResponse(
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyList(),
+                "",
+                false,
+                ""
+            )
+    }
```

Comment:
I'm not crazy about creating this dummy object just to allow a different return mode from that function. How about using `null` instead?

Reply:
I expanded on this idea a bit in https://github.com/dkichler/firefly-plaid-connector-2/pull/1

Reply:
> I'm not crazy about creating this dummy object just to allow a different return mode from that function. How about using null instead?

Coming from Scala, I tend to naturally shy away from `null` - I can give it a shot but I think the overall changes wil

My goal here was to achieve the behaviour with as few changes as possible  - there are no tests covering this piece of code and so testing was limited to adhoc testing against an actual instance.

Admittedly, this approach was my second preference to throwing the exception from `executeTransactionSyncRequest` and catching it upstream, but of course that is complicated by the coroutine boundary.  Handling internal to the method and using an "empty" result was an easy way to minimize the changes. 

> I expanded on this idea a bit in https://github.com/dkichler/firefly-plaid-connector-2/pull/1

It look's like this patch also includes a some additiona refactor for behaviour over and above the intent of this PR.  

I would propose that if you're opposed to the "empty" response, I can refactor this piece to use null (I might suggest Option, but I see Kotlin doesn't have it natively), and you can post the remainder of your suggestion as a standalone PR to adjust cursor catchup behaviour.

Reply:
Swapping in `null` for the empty response wasn't too bad - changes are still pretty minimal.